web/src/content/posts/building-a-new-docs-site-with-rsc.mdx

```diff
@@ -0,0 +1,218 @@
+
+2. Viewing a doc meant invoking the RSC flow to make a request to a server, where the Markdown would be complied and returned on fly.
+
+3. No pre-converted Markdown-&gt;HTML stored anywhere, no content stored in a database, no pre-processing of docs files at build time. However, Redwood's [service caching](https://redwoodjs.com/docs/services#caching) *could* be used, since we would still be performing a full RSC request and converting Markdown to HTML on the fly to populate the cache. (Service caching will be our recommended RSC performance-enhancement workflow—more delicious dog food!).
+
+
+With those requirements in place, we got to work.
+
+## create-redwood-app
+
+If you start a new Redwood app from scratch today, doing so with the canary release will make RSC available via an experimental setup command (follow our guide at the end of the [RSC walkthrough blog post](https://redwoodjs.com/blog/rsc-now-in-redwoodjs) to learn more):
+
+```bash
+npx -y create-redwood-app@canary -y ~/docwood
+cd ~/docwood
+yarn rw experimental setup-streaming-ssr -f
+yarn rw experimental setup-rsc
+```
+
+This starts you with a barebones RSC app to demonstrate usage, but a couple of minor edits gets us back to a blank canvas.
+
+## Routing
+
+Since all docs would be rendered on the fly we really only needed a single "container" page to display any particular doc. We used a [glob-type route parameter](https://redwoodjs.com/docs/router#glob-type) which would take any parts of the URL and put them into a single prop which we would then parse on the backend to determine what doc to actually render:
+
+```javascript
+<Route path="/docs/{path...}" page={DocsRendererPage} name="docs" />
+```
+
+Thanks to this route, everything is sent to the `<DocsRendererPage>` and it accepts a single prop `path` containing the full path for the doc (styling removed for clarity):
+
+```javascript
+const DocsRendererPage = ({ path }) => {
+  return (
+    <div>
+      <aside>
+        <DocsNavigationCell docPath={path} />
+      </aside>
+      <main>
+        <DocsRendererCell docPath={path} />
+      </main>
+    </div>
+  )
+}
+```
+
+Some examples of how a URL maps to the `path` prop:
+
+* `/docs/index` → `"index"`
+
+* `/docs/quick-start` → `"quick-start"`
+
+* `/docs/reference/deploy/baremetal` → `"reference/deploy/baremetal"`
+
+
+We also needed one more route to catch the case of the URL looking like `/docs` where there were no additional path parts:
+
+```javascript
+<Route path="/docs" page={DocsRendererPage} name="docsHome" />
+```
+
+This means a request for `/docs` still renders the `<DocsRenderPage>` but `path` will be `undefined`. The `<DocsRendererCell>` will take care of figuring our what to do in that case!
+
+This results in the following basic, mostly unstyled layout of a typical doc page:
+
+<p align="center">
+  <img src="https://cdn.hashnode.com/res/hashnode/image/upload/v1713207114582/f75d758a-eb91-4391-86c2-14cb4a41cb04.png" />
+</p>
```

Comment:
I couldn't get the markdown centre syntax to work so I just switched to include the html that was previously used. Can improve over time.